File: Week_7_Day_2.js
class Node {
    constructor(value) {
        this.value = value;
        this.next = null;
    }
}

class SLL {
    constructor() {
        this.head = null;
    }

    isEmpty() { 
        return this.head === null ? true : false
    }

    // when a pointer is to the LEFT of an equal sign, we are CHANGING it
    // when a pointer is to the RIGHT of an equal sign, we are READING it

    // add given node to the head
    // list is empty?
    // list already has nodes?
    addToFront(node) {
        if(this.isEmpty()){
            this.head = node;
        } else {
            node.next = this.head;
            this.head = node;
        }
    }
    
    // create a new node with given data, add it to the head. return void
    addDataToFront(data) {
        var node = new Node(data);
        this.addToFront(node);
    }
}

// instantiate the SLL
var myCoolSLL = new SLL(); //
console.log(myCoolSLL);
// console.log(myCoolSLL.isEmpty());
// creating nodes
var myNode = new Node(55);

// executing methods:
myCoolSLL.addToFront(new Node(55));
myCoolSLL.addToFront(new Node(100));

// myCoolSLL.addToFront(new Node(44));
myCoolSLL.addDataToFront(44);
// myCoolSLL.addToFront(new Node(33));
myCoolSLL.addDataToFront(33);
myCoolSLL.addToFront(new Node(22));
// myCoolSLL.addDataToFront(22);
console.log(myCoolSLL);

File: Week_7_Day_2.py
class Node:
    def __init__(self, value):
        self.value = value
        self.next = None

class SLL:
    def __init__(self):
        self.head = None

    def printSLL(self):
        currentNode = self.head
        if currentNode == None:
            print("Empty SLL")
        while currentNode is not None:
            print(currentNode.value)
            currentNode = currentNode.next
    
    def isEmpty(self):
        # return self.head == None
        return True if self.head == None else False
        # ^ practicing ternary in python
    
    def addToFront(self, node):
        # if(self.head == None):
        #     self.head = node
        # else:
        #     node.next = self.head
        #     self.head = node

        node.next = self.head
        self.head = node
        return self
    
    def addDataToFront(self, data):
        newNode = Node(data)
        newNode.next = self.head
        self.head = newNode
        return self

myCoolSLL = SLL()
# print(myCoolSLL)

myCoolSLL.addDataToFront(10)
# print(myCoolSLL.head.value)